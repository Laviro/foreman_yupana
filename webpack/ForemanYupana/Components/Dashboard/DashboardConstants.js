export const YUPANA_POLLING = 'YUPANA_POLLING';
export const YUPANA_POLLING_START = 'YUPANA_POLLING_START';
export const YUPANA_POLLING_STOP = 'YUPANA_POLLING_STOP';
export const YUPANA_POLLING_ERROR = 'YUPANA_POLLING_ERROR';
export const YUPANA_TAB_CHANGED = 'YUPANA_TAB_CHANGED';
export const YUPANA_REPORTS_DOWNLOAD = 'YUPANA_REPORTS_DOWNLOAD';
export const YUPANA_PROCESS_RESTART = 'YUPANA_PROCESS_RESTART';
export const YUPANA_QUEUE = 'YUPANA_QUEUE';
export const YUPANA_QUEUE_ERROR = 'YUPANA_QUEUE_ERROR';
