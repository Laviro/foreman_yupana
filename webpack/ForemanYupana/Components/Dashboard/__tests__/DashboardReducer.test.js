import { testReducerSnapshotWithFixtures } from 'react-redux-test-utils';

import { YUPANA_POLLING_START, YUPANA_POLLING } from '../DashboardConstants';
import reducer from '../DashboardReducer';
import {
  pollingProcessID,
  logs,
  completed,
  initialState,
  files,
} from '../Dashboard.fixtures';

const fixtures = {
  'should return the initial state': initialState,
  'should handle YUPANA_POLLING_START': {
    action: {
      type: YUPANA_POLLING_START,
      payload: {
        pollingProcessID,
      },
    },
  },
  'should handle YUPANA_POLLING': {
    action: {
      type: YUPANA_POLLING,
      payload: {
        logs,
        completed,
        files,
      },
    },
  },
};

describe('Dashboard reducer', () =>
  testReducerSnapshotWithFixtures(reducer, fixtures));
