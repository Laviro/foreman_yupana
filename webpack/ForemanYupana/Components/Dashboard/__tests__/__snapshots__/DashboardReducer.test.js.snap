// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard reducer should handle YUPANA_POLLING 1`] = `
Object {
  "some-account-ID": Object {
    "uploads": Object {
      "error": null,
      "logs": Array [
        "some-logs...",
      ],
    },
  },
}
`;

exports[`Dashboard reducer should handle YUPANA_POLLING_ERROR 1`] = `
Object {
  "some-account-ID": Object {
    "uploads": Object {
      "error": "some-error",
    },
  },
}
`;

exports[`Dashboard reducer should handle YUPANA_POLLING_START 1`] = `
Object {
  "some-account-ID": Object {
    "activeTab": "generating",
    "pollingProcessID": 1,
  },
}
`;

exports[`Dashboard reducer should handle YUPANA_TAB_CHANGED 1`] = `
Object {
  "some-account-ID": Object {
    "activeTab": "uploads",
  },
}
`;

exports[`Dashboard reducer should return the initial state 1`] = `Object {}`;
